### PR TITLE
Use server constructed relative URL for dashboard level links

### PR DESCRIPTION
**Pull request by philbogle**
Rather than constructing a (non-canonical) URL on client for level links in the dashboard header, rely upon a level url provided by the servers.  This has several advantages:
- It saves ~100ms when navigating to a different level, because we don't need to redirect from the non-canonical URL to the canonical one.
- It simplifies the client code by not duplicating logic about how URLs are constructed.

### CODE REVIEW

File: dashboard/app/models/script_level.rb [hunk 1]
```diff
@@ -22,6 +22,9 @@
 # Joins a Script to a Level
 # A Script has one or more Levels, and a Level can belong to one or more Scripts
 class ScriptLevel < ActiveRecord::Base
+  include LevelsHelper
+  include Rails.application.routes.url_helpers
```
Comment (laurelfan) [inline dashboard/app/models/script_level.rb:26]: oh, this is cool, I didn't know you could just do this in models

**Comment by laurelfan** [comment]
Is this already covered by unit tests? (I suspect not because it changed and nothing failed ;))

**Comment by philbogle** [comment]
Good point, I'll add unit tests. There are UI tests that exercise this functionality indirectly (by clicking on the bubbles) but not unit tests.

**Comment by philbogle** [comment]
@laurelfan I added a couple unit tests for the default and custom route cases, PTAL.

**Comment by laurelfan** [comment]
LGTM :+1:

**Pull request merged**